examples/00-fluent/single_battery_cell_workflow.py

```diff
@@ -0,0 +1,719 @@
+# Obtaining the solution
+# =====================================================================================
+
+solver.settings.solution.initialization.standard_initialize()
+solver.settings.solution.run_calculation.transient_controls.time_step_size = 30
+solver.settings.solution.run_calculation.transient_controls.time_step_count = 100
+solver.settings.solution.run_calculation.calculate()
+
+#######################################################################################
+# Post-processing  setup
+# =====================================================================================
+# Configure contours and vector plot for visualizing the results
+
+# Configure graphics settings
+graphics = solver.settings.results.graphics
+graphics.picture.x_resolution = 650
+graphics.picture.y_resolution = 450
+
+# Define and create contour plots
+contours = solver.settings.results.graphics.contour
+contour_list = [
+    {
+        "name": "contour-phi+",
+        "field": "cathode-potential",
+        "surfaces": ["wall_active"],
+        "file_name": "Single_Battery_Cell_1.png",
+    },
+    {
+        "name": "contour-phi-",
+        "field": "anode-potential",
+        "surfaces": ["wall_active"],
+        "file_name": "Single_Battery_Cell_2.png",
+    },
+    {
+        "name": "contour-phi-passive",
+        "field": "passive-zone-potential",
+        "surfaces": ["tab_n", "tab_p", "wall_n", "wall_p"],
+        "file_name": "Single_Battery_Cell_3.png",
+    },
+    {
+        "name": "contour-temp",
+        "field": "temperature",
+        "surfaces": ["wall_p", "wall_active", "tab_p", "tab_n", "wall_n"],
+        "file_name": "Single_Battery_Cell_4.png",
+    },
+]
+
+# Create, display, and save contour plots
+for contour in contour_list:
+    # Create the contour
+    contours.create(contour["name"])
+    current = contours[contour["name"]]
+    current.field = contour["field"]
+    current.surfaces_list = contour["surfaces"]
+    current.range_options.compute()
+
+    # Set the  view
+    graphics.views.restore_view(view_name="front")
+
+    #  display the  current contour
+    current.display()
+
+    # Save the contour plot as an image
+    graphics.picture.save_picture(file_name=contour["file_name"])
+
+# Create and configure vector plot
+vectors = solver.settings.results.graphics.vector
+vectors.create("vector-current_density")
+vector_plot = vectors["vector-current_density"]
+vector_plot.vector_field = "current-density-j"
+vector_plot.field = "current-magnitude"
+vector_plot.surfaces_list = ["wall_n", "wall_p", "wall_active", "tab_n", "tab_p"]
+vector_plot.options.vector_style = "arrow"
+vector_plot.range_options.compute()
+
+# Set view, display, and save the vector plot image
+graphics.views.restore_view(view_name="front")
+vector_plot.display()
+graphics.picture.save_picture(file_name="Single_Battery_Cell_5.png")
+
+# Save case file
+solver.settings.file.write(file_name="unit_battery.cas.h5", file_type="case")
```

Comment:
I realise that the API contains such general methods as `read()` and `write()` as well as the specific versions. I have a preference for the latter, especially while `file_type` is a string argument.